Sources/Hummingbird/Middleware/MiddlewareModule/_OptionalMiddleware.swift

```diff
@@ -0,0 +1,47 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// A middleware that can handle an optional middleware.
+///
+/// This middleware is useful for situations where you want to optionally unwrap a middleware.
+///
+/// You won't typically construct this middleware directly, but instead will use standard `if`-`else`
+/// statements in a parser builder to automatically build conditional middleware:
+///
+/// ```swift
+/// router.addMiddleware {
+///   if let middleware {
+///     middleware
+///   }
+///   ...
+/// }
+/// ```
+public struct _OptionalMiddleware<M0: MiddlewareProtocol>: MiddlewareProtocol {
```

Comment:
Can you add `@_documentation(visibility: internal)`. This isn't a symbol we need to publish documentation for.